File: generate-graph.ts
import fs from 'fs';

// Generate a random graph
function generateGraph() {
    const numNodes = Math.floor(Math.random() * 10) + 1; // Random number of nodes between 1 and 10
    const edges = [];

    // Generate random edges with weights
    for (let i = 1; i <= numNodes; i++) {
        for (let j = i + 1; j <= numNodes; j++) {
            const weight = Math.floor(Math.random() * 10) + 1; // Random weight between 1 and 10
            edges.push([i, j, weight]);
        }
    }

    return edges;
}

// Save graph to CSV file
function saveGraphToCSV(graph: number[][], file: string) {
    const csvData = graph.map(edge => edge.join(',')).join('\n');
    fs.writeFileSync(file, csvData);
}

// Generate and save the graph
for (let i = 1; i <= 100; i++) {
    const graph = generateGraph();
    saveGraphToCSV(graph, `./data/graph${i}.csv`);
}

File: generate_graph.py
import csv
import random

def generate_graph_csv(num_nodes=1000, edge_probability=0.9, negative_weight_probability=0, filename="large_graph_positive_large.csv"):
  """
  Generates a CSV file representing a graph with a specified number of nodes and edges.

  Args:
    num_nodes: The number of nodes in the graph.
    edge_probability: The probability of an edge existing between any two nodes.
    negative_weight_probability: The probability of an edge having a negative weight.
    filename: The name of the CSV file to generate.
  """

  with open(filename, 'w', newline='') as csvfile:
    writer = csv.writer(csvfile)
    writer.writerow(["source", "target", "weight"])  # Header row

    for i in range(num_nodes):
      for j in range(i + 1, num_nodes):
        if random.random() < edge_probability:
          weight = random.randint(1, 10)  # Random weight between 1 and 10
          if random.random() < negative_weight_probability:
            weight = -weight  # Make weight negative with some probability
          writer.writerow([f"Node{i}", f"Node{j}", weight])

if __name__ == "__main__":
  generate_graph_csv()